theano/sandbox/cuda/tests/test_opt.py

```diff
@@ -632,6 +635,43 @@ def cmp(a_shp, b_shp):
     cmp((3, 4), (4,))
 
 
+def test_blocksparse_gpu_gemv_opt():
+    b = tensor.fmatrix()
+    W = tensor.ftensor4()
+    h = tensor.ftensor3()
+    iIdx = tensor.lmatrix()
+    oIdx = tensor.lmatrix()
+
+    o = sparse_block_dot(W, h, iIdx, b, oIdx)
+
+    f = theano.function([W, h, iIdx, b, oIdx], o, mode=mode_with_gpu)
+
+    assert isinstance(f.maker.fgraph.toposort()[-2].op, GpuSparseBlockGemv)
+
+
+def test_blocksparse_gpu_outer_opt():
```

Comment:
@nouiz, @abergeron The optimisation seems to be failing because there is nothing after the op (so no GpuFromHost possible) and for some reason none if the inputs are HostFromGpu. The BlockSparseOuter only appears when we take the gradient of BlockSparseGemv. Do you see anything wrong in my [optimization](https://github.com/bouthilx/Theano/blob/sparse_block_meta_opts/theano/sandbox/cuda/opt.py#L2340)?